README.rst

```diff
@@ -0,0 +1,210 @@
+====================
+CMake Common Scripts
+====================
+This contains a set useful cmake macros
+
+.. contents:: Table of Contents
+   :depth: 3
+
+
+Available Macros
+================
+
+Clang Tidy
+----------
+
+This CMake macro will add clang tidy to a provided target.
+
+Macro:
+
+.. code-block:: cmake
+
+   target_clang_tidy(${PACKAGE_NAME} ${ARGN})
+
+This configures clang-tidy with default arguments to only produce warnings.
+
+.. code-block:: cmake
+
+   target_clang_tidy(${PACKAGE_NAME} ${DEFAULT_CLANG_TIDY_WARNING_ARGS})
+
+This configures clang-tidy with default arguments to only produce errors.
```

Comment:
A more in depth explanation here would be nice. Does this mean that warnings are produced as warnings and errors are produced as errors? On Tesseract we have warnings produced as errors, right?

Reply:
Also, mentioning the macros to disable a line or section would be good.

Reply:
Updated.